pkg/client/introduce/doc.go

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+// Package introduce is responsible for the introduction between agents.
+// The protocol involves at least two participants. A maximum of three participants might be involved.
+// When creating a client a default invitation might be provided. It simplifies the usage when
+// someone wants to do introduction with your client. The simplest way to handle incoming messages (actions) is:
+//	thID, _ := action.Message.ThreadID()
+//	client.Continue(client.InvitationEnvelope(thID))
+//
+// But also there is an exception, when receiving a request.
+// After receiving a request one of the following functions must be executed.
+// - HandleRequest - is used when you do not have a public invitation and
+// it should be provided by one of the introducees.
+// - HandleRequestWithInvitation - is used when you have a public invitation.
+// A bit complicated way to handle incoming messages (actions) is:
+//	if action.Message.Type() == introduce.RequestMsgType {
+//		alice.HandleRequest(action.Message, to, recipient)
+//		OR
+//		alice.HandleRequestWithInvitation(action.Message, inv, to)
+//	}
+//	thID, _ := action.Message.ThreadID()
+//	action.Continue(alice.InvitationEnvelope(thID))
+//
```

Comment:
I suggest adding a whole example that at least starts with registration of the events channel and listening for incoming events.

Reply:
Done!